Apps/phgsgmail/gsgmail_connector.py

```diff
@@ -336,6 +434,75 @@ def _handle_run_query(self, param):
 
         return action_result.set_status(phantom.APP_SUCCESS)
 
+    def _handle_get_email(self, param):
+
+        # Implement the handler here, some basic code is already in
+
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        # Create the credentials with the required scope
+        scopes = ['https://www.googleapis.com/auth/gmail.readonly']
+
+        # Create a service here
+        self.save_progress("Creating GMail service object")
+
+        user_email = param['email']
+
+        ret_val, service = self._create_service(action_result, scopes, "gmail", "v1", user_email)
+
+        if (phantom.is_fail(ret_val)):
+            return action_result.get_status()
+
+        query_string = ""
+        if 'internet_message_id' in param:
+            query_string += " rfc822msgid:{0}".format(self._handle_py_ver_compat_for_input_str(param['internet_message_id']))
+
+        kwargs = {'q': query_string, 'userId': user_email}
+
+        try:
+            messages_resp = service.users().messages().list(**kwargs).execute()
+        except Exception as e:
+            return action_result.set_status(phantom.APP_ERROR, "Failed to get messages", self._get_error_message_from_exception(e))
+
+        messages = messages_resp.get('messages', [])
```

Comment:
suggestion: I think we can encapsulate this into a function `_get_messages`? ( I will leave it up to you)

Reply:
> @joshquerinphantom Could you also look at the Code Check tests failure above? (i.e. https://github.com/phantomcyber/phantom-apps/pull/713/checks?check_run_id=3617815616#step:6:86)

Those checks are fine. The verbose contains markdown content so putting a period at the end will look wrong. The other complaint is about optional parameters (if you look in the code, it checks if the values are there before retrieving them)